lib/iCal.php

```diff
@@ -193,58 +192,49 @@ public static function generate_ical_feed( $post = null ) {
-			$endDate   = date( 'Ymd\THis', strtotime( $event_post->EventEndDate ) );
+			$item = array();
 
-			$tz_startDate = self::wp_date( 'Ymd\THis\Z', strtotime( $event_post->EventStartDate ) );
-			$tz_endDate = self::wp_date( 'Ymd\THis\Z', strtotime( $event_post->EventEndDate ) );
+			$full_format = 'Ymd\THis\Z';
+			$time = (object) array(
+				'start' => self::wp_strtotime( $event_post->EventStartDate ),
+				'end' => self::wp_strtotime( $event_post->EventEndDate ),
+				'modified' => self::wp_strtotime( $event_post->post_modified ),
+				'created' => self::wp_strtotime( $event_post->post_date ),
+			);
 
 			if ( 'yes' == get_post_meta( $event_post->ID, '_EventAllDay', true ) ) {
-				$startDate = substr( $startDate, 0, 8 );
-				$endDate   = substr( $endDate, 0, 8 );
-				// endDate bumped ahead one day to counter iCal's off-by-one error
-				$endDateStamp = strtotime( $endDate );
-				$endDate      = date( 'Ymd', $endDateStamp + 86400 );
-				$type         = 'DATE';
-				$item[] = "DTSTART;VALUE=$type:" . $startDate;
-				$item[] = "DTEND;VALUE=$type:" . $endDate;
-
+				$type = 'DATE';
+				$format = 'Ymd';
 			} else {
 				$type = 'DATE-TIME';
+				$format = $full_format;
+			}
 
-				if ( ! empty( $wp_timezone ) ){
-					$item[] = "DTSTART;TZID=\"{$wp_timezone}\":{$tz_startDate}";
-				}
-				$item[] = 'DTSTART:' . $tz_startDate;
-				$item[] = 'DTSTART:' . $startDate;
+			$tzoned = (object) array(
+				'start' => date( $format, $time->start ),
+				'end' => date( $format, $time->end ),
+				'modified' => date( $full_format, $time->modified ),
+				'created' => date( $full_format, $time->created ),
+			);
 
-				if ( ! empty( $wp_timezone ) ){
-					$item[] = "DTEND;TZID=\"{$wp_timezone}\":{$tz_endDate}";
-				}
-				$item[] = 'DTEND:' . $tz_endDate;
-				$item[] = 'DTEND:' . $endDate;
+			if ( 'DATE' === $type ){
+				$item[] = "DTSTART;VALUE=$type:" . $tzoned->start;
+				$item[] = "DTEND;VALUE=$type:" . $tzoned->end;
+			} else {
+				$item[] = 'DTSTART:' . $tzoned->start;
+				$item[] = 'DTEND:' . $tzoned->end;
 			}
 
-			$item[] = 'DTSTAMP:' . date( 'Ymd\THis', time() );
-			$item[] = 'CREATED:' . str_replace( array( '-', ' ', ':' ), array( '', 'T', '' ), $event_post->post_date );
-			$item[] = 'LAST-MODIFIED:' . str_replace(
-					array(
-						'-',
-						' ',
-						':',
-					), array(
-						'',
-						'T',
-						'',
-					), $event_post->post_modified
-				);
-			$item[] = 'UID:' . $event_post->ID . '-' . strtotime( $startDate ) . '-' . strtotime( $endDate ) . '@' . $blogHome;
-			$item[] = 'SUMMARY:' . $event_post->post_title;
-			$item[] = 'DESCRIPTION:' . str_replace( array( ',', "\n", "\r", "\t" ), array( '\,', '\n', '', '\t' ), strip_tags( $event_post->post_content ) );
+			$item[] = 'DTSTAMP:' . date( $full_format, time() );
+			$item[] = 'CREATED:' . date( $full_format, $tzoned->created );
+			$item[] = 'LAST-MODIFIED:' . date( $full_format, $tzoned->modified );
```

Comment:
These should just use `$tzoned->created` and `$tzoned->modified` directly as those have already been formatted to `$full_format`.  As it is, passing them through `date()` again corrupts their values.